tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -226,16 +232,277 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
       emitNextLine(0)
   }
 
+  // Private API for Rhino
+
+  private[scalajs] object rhinoAPI { // scalastyle:ignore
+    private implicit val GlobalKnowledge: GlobalKnowledge = new Invalidatable {
+      override def invalidate(): Unit = ()
+    }
+
+    def initialize(linkingUnit: LinkingUnit): Unit =
+      startRun(linkingUnit)
+
+    def getHeaderFile(): org.scalajs.core.tools.io.VirtualJSFile =
+      CoreJSLibs.lib(semantics, outputMode)
+
+    def genClassDef(linkedClass: LinkedClass): js.Tree =
+      classEmitter.genClassDef(linkedClass)
+  }
+
+  // Guardian of all global knowledge
+
+  private abstract class InvalidatableBag {
+    def forgetAbout(invalidatable: Invalidatable): Unit
+  }
+
+  private object KnowledgeGuardian { knowledgeGuardian =>
```

Comment:
I have added an additional commit with `KnowledgeGuardian` moved outside of `Emitter`. Is that what you have in mind?